docs/navigation-key.md

```diff
@@ -0,0 +1,26 @@
+---
+id: navigation-key
+title: Using the navigation key
+sidebar_label: Using the navigation key
+---
+
+The `key` parameter comes up repeatedly across different navigation functions. Let's take a look at a summary of its use cases:
+
+### Idempotent navigation with the [`navigate` call](/docs/navigation-actions.html#navigate)
+
+`navigate` function accepts an optional `key` parameter which is used as an identifier for the route to navigate to. If you provide the identifier (as for example in `navigate({ routeName: 'someScreen', params: { someParam: 'someValue' }, key: 'someScreen' })`) and user triggers such navigation action multiple times (eg. by pushing a button that calls the navigation action quickly multiple times in a row), the navigation action will be executed only once.
+
```

Comment:
> Without using the key, calling `navigate({ routeName: 'someScreen' })` twice would result in pushing the screen onto the stack two times

this isn't actually the case anymore in 2.0, this will also be idempotent, you can conceptualize it as a blank key for a route is also a unique key. more in https://github.com/react-navigation/rfcs/blob/master/text/0004-less-pushy-navigate.md